istioctl/cmd/analyze.go

```diff
@@ -16,21 +16,30 @@ package cmd
 
 import (
 	"fmt"
+	"io"
 	"os"
 	"strings"
 
 	"github.com/spf13/cobra"
 
 	"istio.io/istio/galley/pkg/config/analysis/analyzers"
+	"istio.io/istio/galley/pkg/config/analysis/diag"
 	"istio.io/istio/galley/pkg/config/analysis/local"
 	"istio.io/istio/galley/pkg/config/meta/metadata"
 	cfgKube "istio.io/istio/galley/pkg/config/source/kube"
 	"istio.io/istio/pkg/kube"
 )
 
+type FoundAnalyzeIssuesError struct{}
+
+func (f FoundAnalyzeIssuesError) Error() string {
+	return "Found analyzer issues."
+}
+
 var (
-	useKube      bool
-	useDiscovery string
+	useKube               bool
+	useDiscovery          string
+	messageLevelThreshold = diag.Info // messages strictly worse than this will generate an error exit code
```

Comment:
Make this inclusive (>=) instead of exclusive (>). That seems to me like the more standard semantic when setting error levels. e.g. if the user is defining the threshold, they generally care about defining the threshold at the level they want to exclude, not at the level they want to include.

You may already be tracking this for follow-up work (which is fine), but we should add a flag to let the user control this.